overlord/configstate/configcore/runwithstate.go

```diff
@@ -66,6 +66,10 @@ func init() {
 
 	// kernel.{,dangerous-}cmdline-append
 	addWithStateHandler(validateCmdlineAppend, handleCmdlineAppend, &flags{modeenvOnlyConfig: true})
+
+	// experimental.apparmor-prompting
+	addWithStateHandler(nil, doExperimentalApparmorPromptProfileRegeneration, nil)
```

Comment:
My earlier suggestion was to use the existing setup-snap-securoty task so that there is no new task type necessary. 

Is this infeasible? Do you want to sync and discuss this?